app/src/main/java/com/hedvig/app/feature/offer/OfferViewModel.kt

```diff
@@ -374,32 +384,30 @@ class OfferViewModelImpl(
         offerRepository.writeDiscountToCache(quoteIds, data)
 
     override suspend fun triggerOpenChat() {
-        val event = when (chatRepository.triggerFreeTextChat()) {
-            is Either.Left -> Event.Error
-            is Either.Right -> Event.OpenChat
-        }
-        _events.trySend(event)
+        chatRepository.triggerFreeTextChat()
+            .fold(
+                ifLeft = { Event.Error },
+                ifRight = { Event.OpenChat }
+            )
+            .let(_events::trySend)
     }
 
     override fun onOpenQuoteDetails(id: String) {
         viewModelScope.launch {
-            when (val result = getQuoteUseCase.invoke(quoteIds, id)) {
-                GetQuoteUseCase.Result.Error -> {
-                    offerAndLoginStatus.value = OfferAndLoginStatus.Error
-                }
-                is GetQuoteUseCase.Result.Success -> {
+            getQuoteUseCase.invoke(quoteIds, id)
+                .mapLeft { offerAndLoginStatus.value = OfferAndLoginStatus.Error }
+                .map {
                     _events.trySend(
                         Event.OpenQuoteDetails(
                             QuoteDetailItems(
-                                result.quote.displayName,
-                                result.quote.perils.map { PerilItem.Peril(it) },
-                                result.quote.insurableLimits,
-                                result.quote.insuranceTerms
+                                it.displayName,
+                                it.perils.map { PerilItem.Peril(it) },
+                                it.insurableLimits,
+                                it.insuranceTerms
```

Comment:
These would still be nice to be named `quote` instead of `it`. I personally really like avoiding `it` if it's not a trivial one-liner